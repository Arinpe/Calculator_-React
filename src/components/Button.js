import PropTypes from 'prop-types';

export default function Button({
  name, operator, actionBtn, onClick,
}) {
  return (
    <button
      className={`btn ${operator ? 'operatorBtn' : ''} ${
        actionBtn ? 'actionBtn' : ''
      }`}
      value={name}
      onClick={onClick}
      type="button"
    >
      {name}
    </button>
  );
}

Button.defaultProps = {
  operator: false,
  actionBtn: false,
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  operator: PropTypes.bool,
  actionBtn: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};
