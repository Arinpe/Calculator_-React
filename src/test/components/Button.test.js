import React from 'react';
import render from 'react-test-renderer';
import Button from '../../components/Button';

describe('Button', () => {
  const page = render.create(<Button
    name="1"
    onClick={() => {}}
  />);
  test('should render without crashing', () => {
    expect(page).toMatchSnapshot();
  });
});
