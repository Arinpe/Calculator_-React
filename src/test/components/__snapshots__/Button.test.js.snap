// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render without crashing 1`] = `
<button
  className="btn  "
  onClick={[Function]}
  type="button"
  value="1"
>
  1
</button>
`;
